client/my-sites/upgrades/domain-management/edit-contact-info/form-card.jsx

```diff
@@ -406,44 +437,38 @@ class EditContactInfoFormCard extends React.Component {
 	onWhoisUpdate = ( error, data ) => {
 		this.setState( { formSubmitting: false } );
 		if ( data && data.success ) {
-			const { OPENHRS, OPENSRS, WWD } = registrarNames;
+			if ( ! this.requiresConfirmation() ) {
+				this.props.successNotice( this.props.translate(
+					'The contact info has been updated. ' +
+					'There may be a short delay before the changes show up in the public records.'
+				) );
+				return;
+			}
+
+			const currentEmail = this.props.contactInformation.email,
+				strong = <strong />;
 			let message;
 
-			switch ( this.props.selectedDomain.registrar ) {
-				case OPENHRS:
-				case OPENSRS:
-					message = this.props.translate(
-						'The contact info has been updated. ' +
-						'There may be a short delay before the changes show up in the public records.'
-					);
-					break;
-
-				case WWD:
-				default:
-					const oldEmail = this.props.contactInformation.email,
-						newEmail = formState.getFieldValue( this.state.form, 'email' ),
-						strong = <strong />;
-
-					if ( oldEmail === newEmail ) {
-						message = this.props.translate(
-							'An email has been sent to {{strong}}%(email)s{{/strong}}. ' +
-							'Please confirm it to finish this process.',
-							{
-								args: { email: oldEmail },
-								components: { strong }
-							}
-						);
-					} else {
-						message = this.props.translate(
-							'Emails have been sent to {{strong}}%(oldEmail)s{{/strong}} and {{strong}}%(newEmail)s{{/strong}}. ' +
-							'Please ensure they\'re both confirmed to finish this process.',
-							{
-								args: { oldEmail, newEmail },
-								components: { strong }
-							}
-						);
+			if ( this.hasEmailChanged() ) {
+				const newEmail = formState.getFieldValue( this.state.form, 'email' );
+
+				message = this.props.translate(
+					'Emails have been sent to {{strong}}%(oldEmail)s{{/strong}} and {{strong}}%(newEmail)s{{/strong}}. ' +
+					'Please ensure they\'re both confirmed to finish this process.',
+					{
+						args: { oldEmail: currentEmail, newEmail },
```

Comment:
That was my initial idea, but @a8ci18n bot suggested to revert it because we have this string translated in 17 languages. Since the copy did not actually change, it makes sense to keep the placeholder the same so that we don't loose those translations.